gap/examples.gi

```diff
@@ -658,6 +1081,307 @@ function(filt, m, n)
+[IsImmutableDigraph, IsPosInt, IsPosInt],
+function(filt, n, k)
+  local D;
+  D := MakeImmutable(KneserGraphCons(IsMutableDigraph, n, k));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  SetIsRegularDigraph(D, true);
+  SetIsVertexTransitive(D, true);
+  SetIsEdgeTransitive(D, true);
+  if n >= 2 * k then
+    SetChromaticNumber(D, n - 2 * k + 2);
+  else
+    SetChromaticNumber(D, 1);
+  fi;
+  if Float(n) >= ((3 + 5 ^ 0.5) / 2) * Float(k) + 1 then
+    SetIsHamiltonianDigraph(D, true);
+  fi;
+  SetCliqueNumber(D, Int(n / k));
+  return D;
+end);
+
+InstallMethod(KneserGraph, "for two integers", [IsPosInt, IsPosInt],
+{n, k} -> KneserGraphCons(IsImmutableDigraph, n, k));
+
+InstallMethod(KneserGraph, "for a function and two integers",
+[IsFunction, IsPosInt, IsPosInt], KneserGraphCons);
+
+InstallMethod(MycielskiGraphCons, "for IsMutableDigraph and an integer",
+[IsMutableDigraph, IsPosInt],
+function(filt, n)
+  local D, i;
+  if n < 2 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 1,");
+  fi;
+  D := Digraph(IsMutableDigraph, []);
+  DigraphAddVertices(D, 2);
+  DigraphAddEdges(D, [[1, 2], [2, 1]]);
+  for i in [3 .. n] do
+    D := DigraphMycielskian(D);
+  od;
+  return D;
+end);
+
+InstallMethod(MycielskiGraphCons,
+"for IsImmutableDigraph, integer",
+[IsImmutableDigraph, IsPosInt],
+function(filt, n)
+  local D;
+  D := MakeImmutable(MycielskiGraphCons(IsMutableDigraph, n));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  SetChromaticNumber(D, n);
+  SetCliqueNumber(D, 2);
+  SetIsHamiltonianDigraph(D, true);
+  return D;
+end);
+
+InstallMethod(MycielskiGraph, "for an integer", [IsPosInt],
+n -> MycielskiGraphCons(IsImmutableDigraph, n));
+
+InstallMethod(MycielskiGraph, "for a function and an integer",
+[IsFunction, IsPosInt], MycielskiGraphCons);
+
+InstallMethod(OddGraphCons, "for IsMutableDigraph and an integer",
+[IsMutableDigraph, IsInt],
+function(filt, n)
+  if n < 1 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 0,");
+  fi;
+  return KneserGraph(IsMutableDigraph, 2 * n - 1, n - 1);
+end);
+
+InstallMethod(OddGraphCons,
+"for IsImmutableDigraph, integer",
+[IsImmutableDigraph, IsInt],
+function(filt, n)
+  local D;
+  D := MakeImmutable(OddGraphCons(IsMutableDigraph, n));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  SetIsVertexTransitive(D, true);
+  SetIsEdgeTransitive(D, true);
+  SetIsRegularDigraph(D, true);
+  SetIsDistanceRegularDigraph(D, true);
+  SetChromaticNumber(D, 3);
+  return D;
+end);
+
+InstallMethod(OddGraph, "for an integer", [IsInt],
+n -> OddGraphCons(IsImmutableDigraph, n));
+
+InstallMethod(OddGraph, "for a function and an integer",
+[IsFunction, IsInt], OddGraphCons);
+
+InstallMethod(KellerGraphCons, "for IsMutableDigraph and an integer",
+[IsMutableDigraph, IsInt],
+function(filt, n)
+  local D, vertices, i, j;
+  if (not IsInt(n)) or n < 0 then
```

Comment:
This is more of the sort of thing that @flsmith was talking about - `n` is already known to be an integer by this point.
```suggestion
  if n < 0 then
```